### PR TITLE
Update Go version to 1.16.4

### PR DESCRIPTION
**Pull request by michel-laterman**
## What does this PR do?

Update the version of Go being used to 1.16.4

## Checklist

- [x] ~~My code follows the style guidelines of this project~~
- [x] ~~I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] ~~I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates #25346

### CODE REVIEW

**Comment by michel-laterman** [comment]
jenkins run tests

**Comment by kvch** [comment]
The PR LGTM. Could you please open two issues for the two flaky tests and disable them?

**Review by urso — APPROVED** [review]

**Comment by urso** [comment]
Please add an entry to the breaking changes section in `Changelog.next.asciidoc` about requiring OS X 10.14 minimum.

**Comment by michel-laterman** [comment]
jenkins run tests

**Review by kvch — APPROVED** [review]

File: CHANGELOG-developer.next.asciidoc [hunk 1]
```diff
@@ -113,4 +114,5 @@ The list below covers the major changes between 7.0.0-rc2 and master only.
 - Update Go version to 1.15.9. {pull}24442[24442]
 - Update Go version to 1.15.10. {pull}24606[24606]
 - Update Go version to 1.15.12. {pull}25629[25629]
+- Update Go version to 1.16.4. {issue}25346[25346]
```
Comment (kvch) [inline CHANGELOG-developer.next.asciidoc:117]: Please also add a reference to the PR.

**Review by urso — APPROVED** [review]

**Comment by kvch** [comment]
Failing tests are unrelated.

**Comment by urso** [comment]
Looks good. We can merge the PR as is.

**Pull request merged**